### ISSUE TITLE
TileProviderError passed to ErrorEvent has no information for TerrainProvider failure

### ISSUE THREAD

**Issue by thw0rted** [CesiumGS/cesium#8921]
[Sandcastle example](https://sandcastle.cesium.com/#c=fZHPTtwwEMZfZZTTItExqIdKkF1VRb3wpyBYwSUXYw+sVcdejSeJFsSNN+uL1c4CirqoPnn8/T7P53GvGXperugusrdXHHtniWEOgQY4oeS6Fm+v3/UlMWsX3rHZcxMAOvZH0FQrkfWRUsMwYM9fZEVDcaCJrZrUSpyndIgHeKC+fVVNtV+uMEzWSbnlrQNYLRpM7FgobeB2CRd/Xs9G+mXvuAlN2A2N2Rr5Z09BUFs7bs5dEgo56SyLezBfwBjZxJCiJ/TxsQhYauHOSGQMuqXSYofaPTy9ufyF2ejCo3vYjEyB3hJW+1WdZONpUXxlfXftOrKUic0QlVC79jo/UN135jcJmpS2TQBqNbXW1vXg7LypzPglJzFInhJxU4HxOqWsPHTe37gnaqpFrTK/Y/VR25z0sif2elOw1eHifHuIiLXK5edOidHf69zs4yUTbfthFxS6fzpPqacY2x+dSB7ef6ik80Doc266/ws)

Browser: any

Operating System: any

----

Run the Sandcastle and check the Console tab, or better yet the actual devtools console in your browser.  (The Sandcastle console can't handle object logging and the stringify'd version is less enlightening.)

Basically, when the provider fails during the initial load, it emits an error but contains no useful information to tell me what the error was or what I can do about it.  In this case, it's a network error because the VRTheWorld service seems to be offline.  (In case this is actually a temporary outage, you can just change the URL to something else invalid to see the behavior I'm describing.)

**Comment by OmarShehata** [comment]
I think this is the same issue as reported here: https://github.com/CesiumGS/cesium/issues/8313. I've added your Sandcastle there as another test case. 

I'll also open an issue for VRTheWorld not working in hosted Sandcastle to see if we should remove it for now.
